### ISSUE TITLE
Add Support For Terrafirmacraft[enhancemant]

### ISSUE THREAD

**Issue by dexman545** [P3pp3rF1y/AncientWarfare2#169]
Would it be possible to add support for TFC (terrafirmacraft) to the Automation Module? 

What I mean is, could you make it so that the crops and other materials can be processed/harvested by the tools and technologies added by this module? As well as functioning recipes within it. For instance, with the crop farm, it would need to recognize the seeds, the farmland, and and be able to use TFC's fertilizer instead of bonemeal.

**Comment by Dulciphi** [comment]
Just a heads up, you've got a double post of this issue. You might want to close #168 
I tried asking about the same thing for HarvestCraft a while back (see issue #93 ) . It'd be great to use the AW farms with other crops but from Shadow's earlier answer it doesn't sound possible. Even if I was willing or able to code it all manually myself (I'm not) it would still require a hard dependency on the added mod - and that's just not feasible.

**Comment by shadowmage45** [comment]
Negative -- cannot explicitly use items from other mods without adding a hard dependency (or doing retarded hack-arounds).

**Comment by shadowmage45** [comment]
Thinking further on it, I might be able to enable at least some level of compatibility using pure text-references (e.g. through a config file).

Will investigate when I have time, likely won't be until the 2.5 or 2.6 release though

**Comment by Dulciphi** [comment]
awesome. please let us know if there's anything we can do in the meantime

**Comment by shadowmage45** [comment]
Nothing that can really be done in the mean-time.

Basically I need time to sit down with TFC source-code (hopefully they have source available, or this is never going to work?) and figure out how they are handling their plants.  -IF- they have used the base vanilla mechanics (i.e. their crops extend BlockCrop), then there is a good chance I could get something to work.  

If they use completely custom mechanics, it will be much harder (but perhaps still doable), as long as source-code is available. 

If no source code is available, then there is nothing I can do in the long or short-run.

**Comment by shadowmage45** [comment]
A quick look at the TFC source code reveals that this would be a complete bitch to implement.  TFC is not only -not- using the proper vanilla extensibility (e.g. not extending BlockCrop the way they should), but is ALSO not using meta-data for the growth stages.

In short, this will not be possible without creating a dependency (at least a compile-time dependency, which I already have too many of).  TFC would need to rework how their crop blocks are setup in order for me to add non-dependency based compatibility.

In other words -- I cannot support mods that are not doing things properly (proper class/inheritance setup, using metadata where metadata should be used).
